### ISSUE TITLE
[cssom-1] Define `CSSRule.cssText` with `readonly`

### ISSUE THREAD

**Issue by cdoublev** [w3c/csswg-drafts#8778]
Since https://github.com/w3c/csswg-drafts/commit/9f8435ab9f8b2a6ab76e54d2cfd06bec4526ea5a.

  > ```
  > interface CSSRule { 
  >   attribute CSSOMString cssText;
  >   // ...
  > }
  > ```
  > On setting the `cssText` attribute must do nothing.

Would it be different / a problem if [`CSSRule.cssText`](https://w3c.github.io/csswg-drafts/cssom-1/#cssrule) were defined with `readonly`?

**Comment by tabatkins** [comment]
Hm, I thought that readonly attributes threw when you tried to set them, but no, the JS bindings definitely just ignore the set, same as what we specify manually. I wonder what the historical reason was for writing it this way?

/cc @dbaron for maybe some insight?

**Comment by Loirooriol** [comment]
https://webidl.spec.whatwg.org/#dfn-read-only

> It is language binding specific whether assignment is simply disallowed by the language, ignored or an exception is thrown.

For ECMAScript, in https://webidl.spec.whatwg.org/#define-the-attributes the setter will be undefined. So per https://tc39.es/ecma262/#sec-putvalue the behavior will be that it throws in strict mode and it silently fails otherwise.

TL;DR: Marking it as `readonly` would make assignments throw in strict mode.

**Comment by dbaron** [comment]
This is from well before my time; the IDL definition goes back at least to the [first working draft of DOM Level 2](https://www.w3.org/TR/1998/WD-DOM-Level-2-19981228/css.html#Level2-CSS-CSSRule) maintained by the DOM Working Group.  The text defining that the setter does nothing [is substantially newer](https://www.w3.org/TR/2016/WD-cssom-1-20160317/#changes-from-5-december-2013).

**Comment by cdoublev** [comment]
Thanks. I did not think to look at the Web IDL spec for `readonly`.

So it would be different. I assume that making this change this way reduced its impact in terms of backwards compatibility.

I am not requesting a change to `readonly`. I was mostly interested in why implementing a setter that returns nothing is required instead, so I am closing as this question has been answered.
